crypto/default/src/main/java/org/keycloak/crypto/def/DefaultCryptoProvider.java

```diff
@@ -22,13 +22,14 @@
  */
 public class DefaultCryptoProvider implements CryptoProvider {
 
-    private final BouncyCastleProvider bcProvider;
+    private final Provider bcProvider;
 
     private Map<String, Object> providers = new ConcurrentHashMap<>();
 
     public DefaultCryptoProvider() {
         // Make sure to instantiate this only once due it is expensive. And skip registration if already available in Java security providers (EG. due explicitly configured in java security file)
-        BouncyCastleProvider existingBc = (BouncyCastleProvider) Security.getProvider(CryptoConstants.BC_PROVIDER_ID);
+        Provider existingBc = Security.getProvider(CryptoConstants.BC_PROVIDER_ID);
+        System.out.println("Updating BC_Provider");
```

Comment:
Minor note: Is it possible to either remove this line with `System.out.println` or replace with `logger.trace` call? See for example class `KeycloakFipsSecurityProvider` (or many other classes of Keycloak project) on how to add jboss logging to some class.